frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts

```diff
@@ -89,26 +89,49 @@
 
   try {
     // Validate that the submission is of a valid shape.
-    const submission = StorageModeSubmissionStreamDto.parse(JSON.parse(line))
+    const submission = SubmissionStreamDto.parse(JSON.parse(line))
     csvRecord = new CsvRecord(
       submission._id,
       submission.created,
       CsvRecordStatus.Unknown,
       formId,
       hostOrigin,
-      submission.payment,
+      submission.submissionType === SubmissionType.Encrypt
+        ? submission.payment
+        : undefined,
     )
     try {
-      const decryptedObject = formsgSdk.crypto.decrypt(secretKey, {
-        encryptedContent: submission.encryptedContent,
-        verifiedContent: submission.verifiedContent,
-        version: submission.version,
-      })
-
-      if (!decryptedObject) {
-        throw new Error('Invalid decryption')
+      let decryptedSubmission
+      switch (submission.submissionType) {
+        case SubmissionType.Encrypt: {
+          const decryptedObject = formsgSdk.crypto.decrypt(secretKey, {
+            encryptedContent: submission.encryptedContent,
+            verifiedContent: submission.verifiedContent,
+            version: submission.version,
```

Comment:
what's version for?

Reply:
the `Crypto` classes include version, to future-proof also for schema changes. Reason is that the `decrypt` function does shape validation after decryption. So for e.g., it would have been used if we only change the schema without changing the encryption scheme. Then the `Crypto` class can be reused, with the only change being that the shape is validated based on the provided version number